src/library-authoring/data/apiHook.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+import React from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { MeiliSearch } from 'meilisearch';
+
+import { useContentSearchConnection, useContentSearchResults } from '../../search-modal';
+import { getContentLibrary } from './api';
+
+/**
+ * Hook to fetch a content library by its ID.
+ * @param {string} [libraryId] - The ID of the library to fetch.
+ */
+export const useContentLibrary = (libraryId) => (
+  useQuery({
+    queryKey: ['contentLibrary', libraryId],
+    queryFn: () => getContentLibrary(libraryId),
+  })
+);
+
+/**
+ * Hook to fetch the count of components and collections in a library.
+ * @param {string} libraryId - The ID of the library to fetch.
+ * @param {string} searchKeywords - Keywords to search for.
+ */
+export const useLibraryComponentCount = (libraryId, searchKeywords) => {
+  // Meilisearch code to get Collection and Component counts
+  const { data: connectionDetails } = useContentSearchConnection();
+
+  const indexName = connectionDetails?.indexName;
+  const client = React.useMemo(() => {
+    if (connectionDetails?.apiKey === undefined || connectionDetails?.url === undefined) {
+      return undefined;
+    }
+    return new MeiliSearch({ host: connectionDetails.url, apiKey: connectionDetails.apiKey });
+  }, [connectionDetails?.apiKey, connectionDetails?.url]);
+
+  const libFilter = `context_key = "${libraryId}"`;
+
+  const { totalHits: componentCount } = useContentSearchResults({
+    client,
+    indexName,
+    searchKeywords,
+    extraFilter: [libFilter], // ToDo: Add filter for components when collection is implemented
+  });
+
+  const collectionCount = 0; // ToDo: Implement collections count
+
+  return {
+    componentCount,
+    collectionCount,
+  };
+};
```

Comment:
I created this hook to get only the component count for the library. Maybe we should replace this with another instance of the `SearchContext` when we add other filters. @ChrisChV @bradenmacdonald

Reply:
Yes, when we get there what I want to do is move most of the "generic" search code out of `search-modal` into somewhere like `generic/search/` and then we can use it in both `search-modal/SearchModal.tsx` and `src/library-authoring/LibraryHome.tsx` etc.

Reply:
@bradenmacdonald
I was thinking of exporting this directly from our `search-modal` feature. I know the current name is not the best, but it would be our `search-engine` feature that exports both the modal and the provider to use the search engine elsewhere. Maybe rename this feature to `search-engine` to make it explicit.

Or do you think having two features (`search-modal` and `search-engine`) would work best?

Reply:
And, just to confirm: we are moving to use the search index as the primary source for listing content, right?

We might need to move some indexing tasks to run sync because if we add or remove a library, we could refetch before the index update and end up with out of sync data.

Reply:
Yes, exactly. When we make changes to libraries / library content, we can update the search index synchronously.

It's fine to export from search-modal for now, but I do think it's better to separate the library (search manager) from the two use cases: library home UI, and search modal.